docs/superpowers/plans/2026-07-29-observer-node-mode.md

```
@@ -0,0 +1,988 @@
+    }
+```
+
+Add `use rstest::rstest;` to the `node_mode_tests` module's imports. `Serialize` and `Deserialize` come in via the module's `use super::*;`.
+
+- [ ] **Step 2: Run the tests to verify they fail**
+
+```bash
+cargo nextest run -p irys-types node_mode_tests
+```
+
+Expected: `deserialize_peer_reports_the_rename` FAILS — the derived `Deserialize` produces `unknown variant \`Peer\`, expected one of \`Genesis\`, \`Miner\`, \`Observer\``, which contains neither the word "Miner" as guidance nor any mention of the rename. `serde_round_trip` and `deserialize_unknown_mode_lists_the_variants` should already PASS.
```

markdownlint-cli2 (0.23.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)